### PR TITLE
Support of AWS roles / instance profile for Authentication

### PR DESCRIPTION
**Pull request by itsvikramagr**

### CODE REVIEW

**Comment by itsvikramagr** [comment]
@ggeorgiadis, @prithish, @Igorock - Can you please look at this pull requests.

File: src/main/scala/org/apache/spark/sql/kinesis/SparkAWSCredentials.scala [hunk 1]
```diff
@@ -39,6 +39,16 @@ private[kinesis] final case object DefaultCredentials extends SparkAWSCredential
   def provider: AWSCredentialsProvider = new DefaultAWSCredentialsProviderChain
 }
 
+/*
+ * Returns InstanceProfileCredentialsProvider.
+ */
+
+private[kinesis] final case class InstanceProfileCredentials()
```
Comment (ggeorgiadis) [inline src/main/scala/org/apache/spark/sql/kinesis/SparkAWSCredentials.scala]: Shall we make this a case object like DefaultCredentials?

File: src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala [hunk 2]
```diff
@@ -155,10 +155,6 @@ private[kinesis] case class KinesisReader(
         s"Describe Streams") {
         try {
           getAmazonClient.describeStream(describeStreamRequest)
-        } catch {
```
Comment (ggeorgiadis) [inline src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala]: Why remove this catch clause?

Comment (itsvikramagr) [reply src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala]: We are already making describeStream API call within` retryOrTimeout` block which takes care of `LimitExceededException`. If the API call fails with `AccessDeniedException` or any other invalid credentials exception, we should error out and stop the streaming application.  Earlier we were absorbing the exceptions which was given a false impression that streaming job is fine.  See #10 and #12

**Pull request merged**